pkg/agent/plugin/workloadattestor/systemd/systemd_posix.go

```diff
@@ -67,12 +71,29 @@ func (p *Plugin) Attest(ctx context.Context, req *workloadattestorv1.AttestReque
 	}, nil
 }
 
-func getSystemdUnitInfo(ctx context.Context, pid uint) (*DBusUnitInfo, error) {
-	conn, err := dbus.SystemBus()
+func (p *Plugin) DBusConn() (*dbus.Conn, error) {
+	p.dbusMutex.Lock()
+	defer p.dbusMutex.Unlock()
+
+	if p.dbusConn != nil &&
+		p.dbusConn.Connected() {
+		return p.dbusConn, nil
+	}
+
+	conn, err := dbus.ConnectSystemBus()
+	if err != nil {
+		return nil, err
+	}
+	p.dbusConn = conn
+	return p.dbusConn, nil
+}
+
+func getSystemdUnitInfo(ctx context.Context, p *Plugin, pid uint) (*DBusUnitInfo, error) {
+	// Do not close this connection because it is shared and will autoclose on errors
```

Comment:
I'm not sure if this comment is accurate. I would probably mention that the connection is closed when the plugin is unloaded in SPIRE. Something along the lines of:
```suggestion
	// We are not closing the connection here because it's closed when the Close() function is called as part of unloading the plugin.
```